howso/update_cases.amlg

```diff
@@ -218,17 +218,102 @@
+							session session
+						))
+					)
+					case_ids
+				)
+
+				;update derived features if necessary
+				(let
+					(assoc
+						affected_derived_features
+							;any features edited that are the source feature of a non-TS derived feature
+							(filter
+								(lambda
+									(and
+										;not a time-series feature (lag, rate, etc.)
+										(not (contains_index (get !featureAttributes (current_value)) "ts_type"))
+										;generated upon training, otherwise not saved into cases
+										(size (get !featureAttributes [(current_value 1) "auto_derive_on_train"]) )
+										;at least one source feature that affects this derived feature was edited
+										(size (filter
+											(lambda
+												;"list of derived features contains this specific derived feature?""
+												(contains_value (current_value) (current_value 1))
+											)
+											;only iterate over sources features that were edited
+											(keep !sourceToDerivedFeatureMap features)
+										))
+										;the derivation code only uses values from within the same case (not other cases in the series)
+										;(get_all_labels) returns a map of all labels to their values from the code, in this case all feature names to their  offsets,
+										;offsets other than zero indicate deriving from values of another row within a series
+										;tackling this problem for TS data is TODO: 21518
+										(=
+											[0]
+											(values
+												(get_all_labels (parse (get !featureAttributes (list (current_value 1) "auto_derive_on_train" "code"))))
+												(true)
+											)
+										)
+									)
+								)
+								(indices !derivedFeaturesMap)
+							)
+					)
+
+					;this will be a no-op if affected_derived_features is empty
+					(map
+						(lambda
+							(let
+								(assoc
+									derived_feature (current_value 1)
+									necessary_features
+										;the labels (feature names) used in the derivation code
+										(indices (get_all_labels (parse (get !featureAttributes (list (current_value 2) "auto_derive_on_train" "code")))))
+								)
+
+								(declare (assoc
+									necessary_values
+										(map
+											(lambda (retrieve_from_entity (current_value) necessary_features))
+											case_ids
+										)
+								))
+
+								(declare (assoc
+									updated_values
+										;derive new values for each case
+										(call !AddDerivedCodeFeature (assoc
+											feature derived_feature
+											features necessary_features
+											series_data necessary_values
+										))
+								))
+
+								(call !StoreCaseValues (assoc
+									case_values_map (zip case_ids updated_values)
+									label_name derived_feature
+									overwrite (true)
+								))
+							)
+						)
+						affected_derived_features
+					)
+
+					;update features if necessary so that data mass change is correct
+					(if (size affected_derived_features)
```

Comment:
no need to do the size check here, just do the accum below since `affected_derived_features` should be an empty list and will accum correctly ?